=========================================

Live Code 2

Nama  : Maheswara Idopati Akram
Batch : RMT-41

Program ini untuk mengakses dan mengatur data dalam database beverage sales
=========================================

section 1
1. Sebutkan kota yang masuk ke dalam wilayah distribusi penjualan perusahaan ?

In [ ]:
from google.colab import auth
from google.cloud import bigquery

print('Authenticated')

project_id_akun = "ftds-41" # Gunakan GCP Project-ID Anda
client = bigquery.Client(project=project_id_akun)

Authenticated


In [ ]:
df = client.query('''
SELECT * FROM `ftds-hacktiv8-project.phase0_ftds_041_rmt.beverage_sales`
''').to_dataframe()


In [ ]:
df1 = client.query('''
SELECT store_city
FROM `ftds-hacktiv8-project.phase0_ftds_041_rmt.beverage_sales`
GROUP BY store_city
''').to_dataframe()
print(df1)

        store_city
0      CEDAR FALLS
1       MASON CITY
2          CLINTON
3         CLARINDA
4        MUSCATINE
5  WEST DES MOINES


1.  pertimbangan: hanya perlu menggroup kota dimana store berada
2.  insight: data kota yang terdistribusikan produk terlihat di tabel
3. kesimpulan: ada lima kota dimana produk di distribusikan dan nama2nya bisa dilihat di tabel



2.Berapa rentang harga minimum dan maksimum dari unit minumam yang dijual ?

In [ ]:
df2 = client.query('''
SELECT product_name, max(unit_retail_price) - min(unit_retail_price) as unit_price_range
FROM `ftds-hacktiv8-project.phase0_ftds_041_rmt.beverage_sales`
group by product_name
''').to_dataframe()
print(df2)

             product_name  unit_price_range
0         MorningMist Dew              0.00
1      MoonlitMocha Twist             19.01
2    FruitFusion Quencher              7.12
3         VitaWave Splash             15.04
4         SpicedApple Hug              1.73
5     OasisGlow Refresher              1.88
6    Sparkling Bliss Brew              8.10
7        BerryBlitz Chill             12.85
8       CocoCrave Delight             17.11
9        ZenZen Tea Blend              8.35
10     SummerSip Tropical              8.40
11     SootheSip Serenity              1.06
12     CitrusBurst Breeze             52.92
13      PeachyKeen Revive              2.69
14       VelvetChai Charm              3.23
15    VelvetMist Infusion             17.05
16    CaramelCrunch Craze              3.92
17        ChocoMint Bliss             25.84
18       BerryBloom Oasis             11.57
19   RadiantRoot Infusion             43.27
20     FrostyMango Fusion              3.10
21     MorningMist Wakeup       

1.  pertimbangan: harus menemukan nilai maximum dan minimum unit price setiap produk dan menggroup mereka agar tidak dikeluarkan semua 6000+ transaksi tapi hanya 42 produk saja
2.  insight: range harga per unit ter lihat di tabel dan ada beberapa produk dengan range 0 atau harga per unit sama untuk semua jenis kemasan dan paket
3. kesimpulan: ada banyak perbedaan range harga tergantung dengan jenis paket atau kemasan

3. Bandingkan total unit yang terjual dan jumlah transaksi dari masing-masing kota pada tahun 2022 ?

In [ ]:
df3 = client.query('''
SELECT store_city, COUNT(*)*SUM(unit_sold) as total_sold, COUNT(*) as total_transaction
FROM `ftds-hacktiv8-project.phase0_ftds_041_rmt.beverage_sales`
group by store_city
''').to_dataframe()
print(df3)

        store_city  total_sold  total_transaction
0      CEDAR FALLS    48764745               2129
1       MASON CITY      787932                258
2          CLINTON    97536060               2292
3         CLARINDA     8564020                740
4        MUSCATINE     1917108                366
5  WEST DES MOINES     8778257                229


1.  pertimbangan: menghitung jumlah nama kota keluar dikalikan dengan sum dari unit terjual dari setiap kota
2.  insight: terlihat jumlah unit terjual di setiap kota
3. kesimpulan: clinton tempat pembelian terbanyak dan mason city tersedikit

4. Produk apa yang paling banyak diminati pada tahun 2022 ?

In [ ]:
df4 = client.query('''
SELECT COUNT(*)*SUM(unit_sold) as total_sold
FROM `ftds-hacktiv8-project.phase0_ftds_041_rmt.beverage_sales`
where date >= "2022-01-01" AND date < "2023-01-01" AND


''').to_dataframe()
print(df4)

   total_sold
0    41081220


1.  pertimbangan: menghitung jumlah nama kota keluar dikalikan dengan sum dari unit terjual dari setiap kota
2.  insight: terlihat jumlah produk yang paling banyak dibeli
3. kesimpulan: produk yang paling banyak dibeli berjumlah 41081220

Section 2 - Pandas
Jawablah masing-masing pertanyaan dibawah ini dengan menggunakan Pandas.

1. Buatlah sebuah column baru dengan nama season. Column ini berisikan label yang menunjukkan keterangan musim berdasarkan bulan dengan ketentuan sebagai berikut :

In [ ]:
import pandas as pd